raml-parser-2/src/main/java/org/raml/v2/internal/impl/RamlBuilder.java

```diff
@@ -148,17 +153,31 @@ public Node build(Reader content, ResourceLoader resourceLoader, String resource
         }
     }
 
-    private ResourceLoader addRootRamlResourceLoader(ResourceLoader resourceLoader, String resourceLocation)
+    private ResourceLoader addRootRamlResourceLoaders(ResourceLoader resourceLoader, String resourceLocation)
     {
-        File parentFile = new File(actualPath != null ? actualPath : resourceLocation).getParentFile();
-        if (parentFile != null)
+        String rootRamlPath = getRootPath(actualPath != null ? actualPath : resourceLocation);
+
+        if (!Strings.isNullOrEmpty(rootRamlPath))
         {
-            resourceLoader = new CompositeResourceLoader(new RootRamlResourceLoader(parentFile), resourceLoader);
+            resourceLoader = new CompositeResourceLoader(new RootRamlFileResourceLoader(rootRamlPath),
+                    new RootRamlUrlResourceLoader(rootRamlPath),
+                    resourceLoader);
         }
         this.resourceLoader = resourceLoader;
         return resourceLoader;
     }
 
+    private String getRootPath(String rootRamlFileUrl)
+    {
+        List<String> urlSegments = Lists.newArrayList(rootRamlFileUrl.split("/"));
```

Comment:
We are assuming that the path separator is '/'. I think this will work for URLs and Unix file systems but what about Windows FS?